test/models/profile_test.dart

```diff
@@ -0,0 +1,185 @@
+import 'dart:convert';
+
+import 'package:mockito/mockito.dart';
+import 'package:motis_mitfahr_app/account/models/profile.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+
+import '../util/factories/model_factory.dart';
+import '../util/factories/profile_factory.dart';
+import '../util/factories/profile_feature_factory.dart';
+import '../util/factories/report_factory.dart';
+import '../util/factories/review_factory.dart';
+import '../util/mock_server.dart';
+import '../util/mock_server.mocks.dart';
+
+// Die Klasse UrlProcessor muss zu Beginn jeder Testdatei implementiert werden und die Methode processUrl überschrieben werden
+// Wird die Methode ProcessUrl aufgrufen, wird für den dort definierten Fall (in dem Beispiel client.from('drives').select('driver_id,seats')) die Antwort definiert
+// Die Datenbankabfrage an den Client wird so abgefangen und das gewünschte Ergebnis zurückgegeben.
+// Um herauszufinden welche URL durch die jeweilige Datenbankabfrage generiert wird, einfach den auskommentierten Print-Aufruf in der mockServer.Dart Datei aktivieren
+
+void main() {
+  MockUrlProcessor profileProcessor = MockUrlProcessor();
+
+  //setup muss in jeder Testklasse einmal aufgerufen werden
+  setUp(() async {
+    await MockServer.initialize();
+    MockServer.handleRequests(profileProcessor);
+  });
+
+  group('fullname', () {
+    test('returns empty if neither name nor surname are set', () async {
+      Profile profile = ProfileFactory().generateFake(
+        name: NullableParameter(null),
+        surname: NullableParameter(null),
+      );
+      expect(profile.fullName, '');
+    });
+
+    test('returns surname if name is not set', () async {
+      final profile = ProfileFactory().generateFake(
+        name: NullableParameter(null),
+      );
+      expect(profile.fullName, profile.surname);
+    });
+
+    test('returns name if surname is not set', () async {
+      final profile = ProfileFactory().generateFake(
+        surname: NullableParameter(null),
+      );
+      expect(profile.fullName, profile.name);
+    });
+
+    test('returns name and surname if both are set', () async {
+      final profile = ProfileFactory().generateFake(
+        name: NullableParameter('name'),
+        surname: NullableParameter('surname'),
+      );
+      expect(profile.fullName, 'surname name');
+    });
+  });
+
+  group('isCurrentUser', () {
```

Comment:
what if profile.id == null

Reply:
we _could_ check that but it's not really a relevant case for us - also it does not change anything and does not help our coverage %